agent/hcp/telemetry_provider.go

```diff
@@ -0,0 +1,184 @@
+package hcp
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"regexp"
+	"sync"
+	"time"
+
+	goMetrics "github.com/armon/go-metrics"
```

Comment:
nit: do we need to alias this (and hcpclient, hcpTelemetry below)? I think `metrics` is free (+ looks like the standard in the repo)

double-nit: I like Google's suggestion that aliased imports be named like packages w/o upper-case characters or underscores: https://google.github.io/styleguide/go/decisions#:~:text=including%20the%20prohibition%20on%20the%20use%20of%20underscores%20and%20capital%20letters.

Reply:
fixed!